packages/headless/src/features/recent-queries/recent-queries-slice.ts

```diff
@@ -11,7 +11,10 @@ export const recentQueriesReducer = createReducer(
   (builder) => {
     builder
       .addCase(registerRecentQueries, (state, action) => {
-        state.queries = action.payload.queries;
+        state.queries = action.payload.queries.slice(
```

Comment:
This change is not related to the actual bug. It would be cleaner to have a separate issue for that. If you prefer to keep it in this PR, make sure to document it in the Jira issue so we know later on why this code changed.

Reply:
I'll add it to the jira.
As for the test, I don't know if it's possible to test that it's logged given the test doesn't go through a whole async loop of fetching the search results and the handling of the search analytics actions is done by the `executeSearch` action it's passed to. That part of the code is tested in the search-slice.tests though.